pkg/cmd/adm/restart.go

```diff
@@ -1,157 +1,213 @@
+	//if there is no operator deployment, no need for restart
+	if len(operatorDeploymentList.Items) == 0 {
+		return fmt.Errorf("no operator based deployment found in namespace %s , hence no restart happened", ns)
 	}
+	//Deleting the pods of the operator based deployment  and then checking the status
+	for _, operatorDeployment := range operatorDeploymentList.Items {
+		ctx.Printlnf("Proceeding to delete the Pods of %v", operatorDeployment.Name)
 
-	ctx.Printlnf("The deployment was scaled back to '%d'", originalReplicas)
-	return nil
-}
+		if err := deleteAndWaitForPods(ctx, cl, operatorDeployment); err != nil {
+			return err
+		}
 
-func restartHostOperator(ctx *clicontext.CommandContext, hostClient runtimeclient.Client, hostNamespace string) error {
-	deployments := &appsv1.DeploymentList{}
-	if err := hostClient.List(context.TODO(), deployments,
-		runtimeclient.InNamespace(hostNamespace),
-		runtimeclient.MatchingLabels{"olm.owner.namespace": "toolchain-host-operator"}); err != nil {
-		return err
+		ctx.Printlnf("Checking the status of the deleted pod's deployment %v", operatorDeployment.Name)
+		//check the rollout status
+		if err := checker(ctx, f, ioStreams, "kubesaw-control-plane=kubesaw-controller-manager"); err != nil {
+			return err
+		}
+	}
+
+	if len(nonOperatorDeploymentList.Items) == 0 {
+		// if there are no non-operator deployments
+		ctx.Printlnf("No Non-operator deployment found in namespace %s, hence no restart happened", ns)
+		return nil
 	}
-	if len(deployments.Items) != 1 {
-		return fmt.Errorf("there should be a single deployment matching the label olm.owner.namespace=toolchain-host-operator in %s ns, but %d was found. "+
-			"It's not possible to restart the Host Operator deployment", hostNamespace, len(deployments.Items))
+	for _, nonOperatorDeployment := range nonOperatorDeploymentList.Items {
+		if nonOperatorDeployment.Name != "autoscaling-buffer" {
+			ctx.Printlnf("Proceeding to restart the non-operator deployment %v", nonOperatorDeployment.Name)
+
+			if err := restarter(ctx, nonOperatorDeployment, f, ioStreams); err != nil {
+				return err
+			}
+			//check the rollout status
+			ctx.Printlnf("Checking the status of the rolled out deployment %v", nonOperatorDeployment.Name)
+			if err := checker(ctx, f, ioStreams, "toolchain.dev.openshift.com/provider=codeready-toolchain"); err != nil {
+				return err
+			}
+			return nil
+		}
+		ctx.Printlnf("Found only autoscaling-buffer deployment in namespace %s , which is not required to be restarted", ns)
 	}
 
-	return restartDeployment(ctx, hostClient, hostNamespace, deployments.Items[0].Name)
+	return nil
 }
 
-func printExistingDeployments(term ioutils.Terminal, cl runtimeclient.Client, ns string) error {
-	deployments := &appsv1.DeploymentList{}
-	if err := cl.List(context.TODO(), deployments, runtimeclient.InNamespace(ns)); err != nil {
+func deleteAndWaitForPods(ctx *clicontext.CommandContext, cl runtimeclient.Client, deployment appsv1.Deployment) error {
```

Comment:
the function doesn't wait for anything, does it?
```suggestion
func deleteDeploymentPods(ctx *clicontext.CommandContext, cl runtimeclient.Client, deployment appsv1.Deployment) error {
```

Reply:
i thought of putting a sleep but unsure

Reply:
you mean to make some time gap between deleting the pods and waiting for the status? yeah, why not

Reply:
added a sleep timer